WordPress/Classes/ViewRelated/Post/PostListViewController.m

```diff
@@ -0,0 +1,1319 @@
+                                                                                   metrics:nil
+                                                                                     views:views]];
+    [self.searchWrapperView addConstraints:[NSLayoutConstraint constraintsWithVisualFormat:@"V:[searchBar]|"
+                                                                                   options:0
+                                                                                   metrics:nil
+                                                                                     views:views]];
+}
+
+- (void)configureSearchBarPlaceholder
+{
+    // Adjust color depending on where the search bar is being presented.
+    UIColor *placeholderColor = [UIDevice isPad] ? [WPStyleGuide grey] : [WPStyleGuide wordPressBlue];
+    NSString *placeholderText = NSLocalizedString(@"Search", @"Placeholder text for the search bar on the post screen.");
+    NSAttributedString *attrPlacholderText = [[NSAttributedString alloc] initWithString:placeholderText attributes:[WPStyleGuide defaultSearchBarTextAttributes:placeholderColor]];
+    [[UITextField appearanceWhenContainedIn:[UISearchBar class], [self class], nil] setAttributedPlaceholder:attrPlacholderText];
+}
+
+- (void)configureSearchWrapper
+{
+    self.searchWrapperView.backgroundColor = [WPStyleGuide wordPressBlue];
+}
+
+
+#pragma mark - Actions
+
+- (IBAction)refresh:(id)sender
+{
+    [self syncItemsWithUserInteraction:YES];
+}
+
+- (IBAction)handleAddButtonTapped:(id)sender
+{
+    [self createPost];
+}
+
+- (IBAction)handleSearchButtonTapped:(id)sender
+{
+    [self toggleSearch];
+}
+
+- (IBAction)handleAuthorFilterChanged:(id)sender
+{
+    if (self.authorsFilter.selectedSegmentIndex == PostAuthorFilterMine) {
+        [self setCurrentPostAuthorFilter:PostAuthorFilterMine];
+    } else {
+        [self setCurrentPostAuthorFilter:PostAuthorFilterEveryone];
+    }
+}
+
+- (void)didTapNoResultsView:(WPNoResultsView *)noResultsView
+{
+    if ([self currentPostListFilter].filterType == PostListStatusFilterScheduled) {
+        NSInteger index = [self indexForFilterWithType:PostListStatusFilterDraft];
+        [self setCurrentFilterIndex:index];
+        return;
+    }
+    [self createPost];
+}
+
+- (IBAction)didTapFilterButton:(id)sender
+{
+    if (self.postFilterPopoverController) {
+        return;
+    }
+    [self displayFilters];
+}
+
+
+#pragma mark - Syncing
+
+- (void)automaticallySyncIfAppropriate
+{
+    // Only automatically refresh if the view is loaded and visible on the screen
+    if (self.isViewLoaded == NO || self.view.window == nil) {
+        DDLogVerbose(@"View is not visible and will not check for auto refresh.");
+        return;
+    }
+
+    // Do not start auto-sync if connection is down
+    WordPressAppDelegate *appDelegate = [WordPressAppDelegate sharedInstance];
+    if (appDelegate.connectionAvailable == NO) {
+        [self configureNoResultsView];
```

Comment:
I mentioned this a while back in #1148:

> The "no content" messages should never be displayed until we have asked the server for the data and confirmed there's actually no content to display. I can imagine someone freaking out on a slow connection "What do you mean I haven't created any posts?"

I know it was behaving like that already, but I don't love seeing bad patterns like this making it into new view controllers.